python/ql/src/experimental/semmle/python/frameworks/SqlAlchemy.qll

```diff
@@ -0,0 +1,97 @@
+/**
+ * Provides classes modeling security-relevant aspects of the 'SqlAlchemy' package.
+ * See https://pypi.org/project/SQLAlchemy/.
+ */
+
+private import python
+private import semmle.python.dataflow.new.DataFlow
+private import semmle.python.dataflow.new.TaintTracking
+private import semmle.python.ApiGraphs
+private import semmle.python.Concepts
+
+private module SqlAlchemy {
+  /**
+   * An instantization of a SqlAlchemy Session object.
+   * See https://docs.sqlalchemy.org/en/14/orm/session_api.html#sqlalchemy.orm.Session and
+   * https://docs.sqlalchemy.org/en/14/orm/session_api.html#sqlalchemy.orm.sessionmaker
+   */
+  private class SqlAlchemySessionInstance extends API::Node {
+    SqlAlchemySessionInstance() {
+      this in [
+          API::moduleImport("sqlalchemy.orm").getMember("Session").getReturn(),
+          API::moduleImport("sqlalchemy.orm").getMember("sessionmaker").getReturn().getReturn()
+        ]
+    }
+
+    override string toString() { result = "Use of SqlAlchemy Session instantization" }
+  }
+
+  /**
+   * An instantization of a SqlAlchemy Engine object.
+   * See https://docs.sqlalchemy.org/en/14/core/engines.html#sqlalchemy.create_engine
+   */
+  private class SqlAlchemyEngineInstance extends API::Node {
+    SqlAlchemyEngineInstance() {
+      this = API::moduleImport("sqlalchemy").getMember("create_engine").getReturn()
+    }
+
+    override string toString() { result = "Use of SqlAlchemy create_engine member" }
```

Comment:
I'm not 100% the best wording practices here so if this needs to be changed let me know!

Reply:
Usually we don't build classes for this, but instead use a predicate, such as 

```codeql

/** Gets an instance of ... */
getSqlAlchemyEngineInstance() {
  result = API::moduleImport("sqlalchemy").getMember("create_engine").getReturn()
}
```

and I think doing so in this case would also make things a bit more clean.